apps/web/components/GroupedTable.tsx

```diff
@@ -555,28 +557,41 @@ function ExpandedRows({
 
   return (
     <>
-      {data.threads.map((thread) => (
-        <TableRow key={thread.id} className="bg-muted/50">
-          <TableCell className="py-3">
-            <ViewEmailButton threadId={thread.id} messageId={thread.id} />
-          </TableCell>
-          <TableCell className="py-3">
-            <Link
-              href={getGmailUrl(thread.id, userEmail)}
-              target="_blank"
-              className="hover:underline"
-            >
-              {thread.messages[0].headers.subject}
-            </Link>
-          </TableCell>
-          <TableCell className="py-3">
-            {decodeSnippet(thread.messages[0].snippet)}
-          </TableCell>
-          <TableCell className="text-nowrap py-3">
-            {formatShortDate(new Date(thread.messages[0].headers.date))}
-          </TableCell>
-        </TableRow>
-      ))}
+      {data.threads.map((thread) => {
+        const firstMessage = thread.messages[0];
+        // Handle both Google (with headers) and Microsoft (direct properties) message structures
+        const subject =
+          "headers" in firstMessage
+            ? firstMessage.headers.subject
+            : firstMessage.subject;
+        const date =
+          "headers" in firstMessage
+            ? firstMessage.headers.date
+            : firstMessage.receivedAt;
```

Comment:
i think we should be returning a unified format on the server rather than lots of one off checks like this.

eg. set message.date, and message.subject, so that you don't need any special logic in this part of the code